addons/website_sale/tests/test_ecommerce_visibility.py

```diff
@@ -0,0 +1,44 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo.tests import tagged
+
+from odoo.addons.base.tests.common import HttpCaseWithUserDemo
+from odoo.addons.website_sale.tests.common import WebsiteSaleCommon
+
+
+@tagged('post_install', '-at_install')
+class TestWSaleEcommerceAccess(HttpCaseWithUserDemo, WebsiteSaleCommon):
+
+    def test_ecommerce_access_public_user(self):
+
+        # By default, everyone has access to ecommerce
+        self.assertTrue(self.website.with_user(self.public_user).has_ecommerce_access())
+        self.website.ecommerce_access = 'logged_in'
+        self.assertFalse(self.website.with_user(self.public_user).has_ecommerce_access())
+
+    def test_has_ecommerce_access_logged_user(self):
```

Comment:
Consistency between tests. Or do the opposite for other inconsistent methods.
```suggestion
    def test_ecommerce_access_logged_user(self):
```